Source/WebCore/history/CachedPage.cpp

```diff
@@ -58,9 +58,9 @@ DEFINE_DEBUG_ONLY_GLOBAL(WTF::RefCountedLeakCounter, cachedPageCounter, ("Cached
 CachedPage::CachedPage(Page& page)
     : m_page(page)
     , m_expirationTime(MonotonicTime::now() + page.settings().backForwardCacheExpirationInterval())
-    , m_cachedMainFrame(makeUnique<CachedFrame>(page.mainFrame()))
+    , m_cachedMainFrame(dynamicDowncast<LocalFrame>(page.mainFrame()) ? makeUnique<CachedFrame>(*dynamicDowncast<LocalFrame>(page.mainFrame())) : nullptr)
```

Comment:
The first "dynamicDowncast" should be "is" and the second should be just "downcast" and I think the * is unneeded with that change.